temba/api/tests.py

```diff
@@ -3166,8 +3183,15 @@ def test_send(self):
                 self.assertEquals('1234567890', msg.external_id)
                 self.assertTrue(msg.sent_on)
 
-                self.clear_cache()
+    def test_send_error(self):
+        # This tests a send error.
+        joe = self.create_contact("Joe", number="+250788383383", twitter="joe1981")
+        bcast = joe.send("Test message", self.admin, trigger_send=False)
+
+        # our outgoing message
+        msg = bcast.get_messages()[0]
```

Comment:
I thought about it, but there are lots of differences between the various `self.create_contact()` and `joe.send()` calls. Factoring them out into a method on each test class doesn't really buy very much. For the tests already had them in `.setUp()` I moved them to `.setup_channel()`, but I'm not sure they really belong there. (If they do, the method should probably be renamed.)